src/test/regress/expected/gp_parallel.out

```diff
@@ -1847,6 +1847,150 @@ abort;
+explain(costs off) create table ctas_ao using ao_row as select sum(a.c2) as c2, avg(b.c1) as c1 from t_p2 a join t_p2 b on a.c1 = b.c1 distributed by(c2);
+                       QUERY PLAN                       
+--------------------------------------------------------
+ Redistribute Motion 1:3  (slice1; segments: 1)
+   Hash Key: (sum(a.c2))
+   ->  Finalize Aggregate
+         ->  Gather Motion 3:1  (slice2; segments: 3)
+               ->  Partial Aggregate
+                     ->  Hash Join
+                           Hash Cond: (a.c1 = b.c1)
+                           ->  Seq Scan on t_p2 a
+                           ->  Hash
+                                 ->  Seq Scan on t_p2 b
+ Optimizer: Postgres query optimizer
+(11 rows)
+
+explain(costs off) create table ctas_aoco using ao_column as select sum(a.c2) as c2, avg(b.c1) as c1 from t_p2 a join t_p2 b on a.c1 = b.c1 distributed by(c2);
+                       QUERY PLAN                       
+--------------------------------------------------------
+ Redistribute Motion 1:3  (slice1; segments: 1)
+   Hash Key: (sum(a.c2))
+   ->  Finalize Aggregate
+         ->  Gather Motion 3:1  (slice2; segments: 3)
+               ->  Partial Aggregate
+                     ->  Hash Join
+                           Hash Cond: (a.c1 = b.c1)
+                           ->  Seq Scan on t_p2 a
+                           ->  Hash
+                                 ->  Seq Scan on t_p2 b
+ Optimizer: Postgres query optimizer
+(11 rows)
+
+set local enable_parallel = on;
+explain(costs off) create table ctas_ao using ao_row as select sum(a.c2) as c2, avg(b.c1) as c1 from t_p2 a join t_p2 b on a.c1 = b.c1 distributed by(c2);
+                           QUERY PLAN                            
+-----------------------------------------------------------------
+ Redistribute Motion 1:3  (slice1; segments: 1)
+   Hash Key: (sum(a.c2))
+   ->  Finalize Aggregate
+         ->  Gather Motion 6:1  (slice2; segments: 6)
+               ->  Partial Aggregate
+                     ->  Parallel Hash Join
+                           Hash Cond: (a.c1 = b.c1)
+                           ->  Parallel Seq Scan on t_p2 a
+                           ->  Parallel Hash
+                                 ->  Parallel Seq Scan on t_p2 b
+ Optimizer: Postgres query optimizer
+(11 rows)
+
+explain(costs off) create table ctas_aoco using ao_column as select sum(a.c2) as c2, avg(b.c1) as c1 from t_p2 a join t_p2 b on a.c1 = b.c1 distributed by(c2);
+                           QUERY PLAN                            
+-----------------------------------------------------------------
+ Redistribute Motion 1:3  (slice1; segments: 1)
+   Hash Key: (sum(a.c2))
+   ->  Finalize Aggregate
+         ->  Gather Motion 6:1  (slice2; segments: 6)
+               ->  Partial Aggregate
+                     ->  Parallel Hash Join
+                           Hash Cond: (a.c1 = b.c1)
+                           ->  Parallel Seq Scan on t_p2 a
```

Comment:
Please try:
CREATE TABLE tbl(a int) DISTRIBUTED randomly;
CREATE materialized view v as SELECT * FROM tbl;
Refresh v;

Reply:
@my-ship-it  The conclusion is: **parallel scan into materialized view is ok, but as before(Motion(N:1)issue, and no startup cost for parallel plan), it's slower than a non-parallel scan.** A plain parallel scan is typically pointless.
But the result is right.

```sql
gpadmin=# create table t1(c1 int, c2 int) distributed randomly;
CREATE TABLE
gpadmin=# insert into t1 select i, i+1 from generate_series(1, 1000000) i;
INSERT 0 1000000
gpadmin=# CREATE materialized view v as SELECT * FROM t1;
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause. Creating a NULL policy entry.
SELECT 1000000
analyze t1;
set enable_parallel=off;
set optimizer=off;
gpadmin=# refresh materialized view v;
REFRESH MATERIALIZED VIEW
Time: 487.095 ms
gpadmin=# select sum(c1), sum(c2) from v;
     sum      |     sum
--------------+--------------
 500000500000 | 500001500000
(1 row)

Time: 152.428 ms

set enable_parallel=on;
set max_parallel_workers_per_gather = 8;
set min_parallel_table_scan_size = 0;
explain select * from t1;
                                       QUERY PLAN
----------------------------------------------------------------------------------------
 Gather Motion 21:1  (slice1; segments: 21)  (cost=0.00..11319.38 rows=1000000 width=8)
   ->  Parallel Seq Scan on t1  (cost=0.00..843.19 rows=47619 width=8)
 Optimizer: Postgres query optimizer
(3 rows)
gpadmin=# refresh materialized view v;
REFRESH MATERIALIZED VIEW
Time: 877.829 ms
gpadmin=# select sum(c1), sum(c2) from v;
     sum      |     sum
--------------+--------------
 500000500000 | 500001500000
(1 row)

Time: 41.330 ms
```